crates/turborepo-lib/src/turbo_json/parser.rs

```diff
@@ -0,0 +1,576 @@
+        // range of the map in the source text.
+        _: TextRange,
+        _name: &str,
+        diagnostics: &mut Vec<DeserializationDiagnostic>,
+    ) -> Option<Self::Output> {
+        let mut result = ConfigurationOptions::default();
+        for (key, value) in members.flatten() {
+            // Try to deserialize the key as a string.
+            // We use `Text` to avoid an heap-allocation.
+            let Some(key_text) = Text::deserialize(&key, "", diagnostics) else {
+                // If this failed, then pass to the next key-value pair.
+                continue;
+            };
+            match key_text.text() {
+                "apiUrl" | "apiurl" | "ApiUrl" | "APIURL" => {
+                    if let Some(api_url) = String::deserialize(&value, &key_text, diagnostics) {
+                        result.api_url = Some(api_url);
+                    }
+                }
+                "loginUrl" | "loginurl" | "LoginUrl" | "LOGINURL" => {
+                    if let Some(login_url) = String::deserialize(&value, &key_text, diagnostics) {
+                        result.login_url = Some(login_url);
+                    }
+                }
+                "teamSlug" | "teamslug" | "TeamSlug" | "TEAMSLUG" => {
+                    if let Some(team_slug) = String::deserialize(&value, &key_text, diagnostics) {
+                        result.team_slug = Some(team_slug);
+                    }
+                }
+                "teamId" | "teamid" | "TeamId" | "TEAMID" => {
+                    if let Some(team_id) = String::deserialize(&value, &key_text, diagnostics) {
+                        result.team_id = Some(team_id);
+                    }
+                }
+                "token" => {
+                    if let Some(token) = String::deserialize(&value, &key_text, diagnostics) {
+                        result.token = Some(token);
+                    }
+                }
+                "signature" => {
+                    if let Some(signature) = bool::deserialize(&value, &key_text, diagnostics) {
+                        result.signature = Some(signature);
+                    }
+                }
+                "preflight" => {
+                    if let Some(preflight) = bool::deserialize(&value, &key_text, diagnostics) {
+                        result.preflight = Some(preflight);
+                    }
+                }
+                "timeout" => {
+                    if let Some(timeout) = u64::deserialize(&value, &key_text, diagnostics) {
+                        result.timeout = Some(timeout);
+                    }
+                }
+                "enabled" => {
+                    if let Some(enabled) = bool::deserialize(&value, &key_text, diagnostics) {
+                        result.enabled = Some(enabled);
+                    }
+                }
+                unknown_key => {
+                    const ALLOWED_KEYS: &[&str] = &[
+                        "apiUrl",
```

Comment:
If we're going with handwritten strings for this, then I think putting them in constants so they can be shared will at least prevent some typo-related errors.